src/CtxHooks.js

```diff
@@ -0,0 +1,97 @@
+/* eslint react/prop-types: 0 */
+// @flow
+import * as React from 'react';
+
+class CtxStore {
+  state = {}
+  subs = [];
+
+  subscribe(handler: Function) {
+    const wrapper = () => {
+      handler(this.state);
+    };
+
+    this.subs.push(wrapper);
+
+    const unsub = () => {
+      const index = this.subs.indexOf(handler);
+
+      if (index !== -1) {
+        this.subs.splice(index, 1);
+      }
+    };
+
+    return unsub;
+  }
+
+  replaceState(newState: any) {
+    this.state = newState;
+  }
+
+  notifyAll() {
+    this.subs.forEach(sub => sub());
+  }
+}
+
+function createUseSelector(Context) {
+  const execSelector = (value, selector) => {
+    if (typeof selector === 'function') {
+      return selector(value);
+    } else if (selector === null || selector === undefined) {
+      return value;
+    } else {
+      throw new Error('Invalid selector');
```

Comment:
Could be more clear with "Expected the selector to be a function or null/undefined"